lib/iris/tests/unit/experimental/regrid/test__CurvilinearRegridder.py

```diff
@@ -35,76 +40,152 @@
+        self.src_grid = lat_lon_cube()
         self.bad = np.ones((3, 4))
-        self.weights = np.ones(self.ok.shape, self.ok.dtype)
+        self.weights = np.ones(self.src_grid.shape, self.src_grid.dtype)
 
     def test_bad_src_type(self):
         with self.assertRaisesRegexp(TypeError, "'src_grid_cube'"):
-            Regridder(self.bad, self.ok, self.weights)
+            Regridder(self.bad, self.src_grid, self.weights)
 
     def test_bad_grid_type(self):
         with self.assertRaisesRegexp(TypeError, "'target_grid_cube'"):
-            Regridder(self.ok, self.bad, self.weights)
+            Regridder(self.src_grid, self.bad, self.weights)
 
 
 @tests.skip_data
 class Test___call__(tests.IrisTest):
     def setUp(self):
-        self.func = ('iris.experimental.regrid.'
-                     'regrid_weighted_curvilinear_to_rectilinear')
-        self.ok = global_pp()
-        y = self.ok.coord('latitude')
-        x = self.ok.coord('longitude')
-        self.ok.remove_coord('latitude')
-        self.ok.remove_coord('longitude')
-        self.ok.add_aux_coord(y, 0)
-        self.ok.add_aux_coord(x, 1)
-        self.weights = np.ones(self.ok.shape, self.ok.dtype)
+        self.func_setup = (
+            'iris.experimental.regrid.'
+            '_regrid_weighted_curvilinear_to_rectilinear__prepare')
+        self.func_operate = (
+            'iris.experimental.regrid.'
+            '_regrid_weighted_curvilinear_to_rectilinear__perform')
+        # Define a test source grid and target grid, basically the same.
+        self.src_grid = global_pp()
+        self.tgt_grid = global_pp()
+        # Modify the names so we can tell them apart.
+        self.src_grid.rename('src_grid')
+        self.tgt_grid.rename('TARGET_GRID')
+        # Replace the source-grid x and y coords with equivalent 2d versions.
+        x_coord = self.src_grid.coord('longitude')
+        y_coord = self.src_grid.coord('latitude')
+        nx, = x_coord.shape
+        ny, = y_coord.shape
+        xx, yy = np.meshgrid(x_coord.points, y_coord.points)
+        self.src_grid.remove_coord(x_coord)
+        self.src_grid.remove_coord(y_coord)
+        x_coord_2d = AuxCoord(xx,
+                              standard_name=x_coord.standard_name,
+                              units=x_coord.units,
+                              coord_system=x_coord.coord_system)
+        y_coord_2d = AuxCoord(yy,
+                              standard_name=y_coord.standard_name,
+                              units=y_coord.units,
+                              coord_system=y_coord.coord_system)
+        self.src_grid.add_aux_coord(x_coord_2d, (0, 1))
+        self.src_grid.add_aux_coord(y_coord_2d, (0, 1))
+        self.weights = np.ones(self.src_grid.shape, self.src_grid.dtype)
+        # Define an actual, dummy cube for the internal partial result, so we
+        # can do a cubelist merge on it, which is too complicated to mock out.
+        self.mock_slice_result = Cube([1])
```

Comment:
A little thing - with the `mock` library flying about in these tests you could rename this cube to `fake_slice_result` to avoid overloading the work 'mock'...